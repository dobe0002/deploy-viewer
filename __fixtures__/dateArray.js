export default [
  new Date(2019, 10, 1), // Fri Nov 01 2019 00: 00: 00 GMT - 0500(Central Daylight Time),
  new Date(2019, 10, 2), // Sat Nov 02 2019 00:00:00 GMT-0500 (Central Daylight Time),
  new Date(2019, 10, 3), // Sun Nov 03 2019 00:00:00 GMT-0500 (Central Daylight Time),
  new Date(2019, 10, 4), // Mon Nov 04 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 5), // Tue Nov 05 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 6), // Wed Nov 06 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 7), // Thu Nov 07 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 8), // ri Nov 08 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 9), // Sat Nov 09 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 10), // Sun Nov 10 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 11), // Mon Nov 11 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 12), // Tue Nov 12 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 13), // Wed Nov 13 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 14), // Thu Nov 14 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 15), // Fri Nov 15 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 16), // Sat Nov 16 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 17), // Sun Nov 17 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 18), // Mon Nov 18 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 19), // Tue Nov 19 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 20), // Wed Nov 20 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 21), // Thu Nov 21 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 22), // Fri Nov 22 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 23), // Sat Nov 23 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 24), // Sun Nov 24 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 25), // Mon Nov 25 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 26), // Tue Nov 26 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 27), // Wed Nov 27 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 28), // Thu Nov 28 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 29), // Fri Nov 29 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 30), // Sat Nov 30 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 11, 1), // Sun Dec 01 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 11, 2) // Mon Dec 02 2019 00:00:00 GMT-0600 (Central Standard Time)
];
