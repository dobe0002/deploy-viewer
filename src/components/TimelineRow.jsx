import React from 'react';
import PropTypes from 'prop-types';
import { compareAsc } from 'date-fns';
import filter from 'lodash/filter';
import Status from './Status';

export default class TimelineRow extends React.Component {
  buildRow() {
    const { dateArray, appStatusArray, appTitle } = this.props;
    return dateArray.map(date => {
      const status = filter(
        appStatusArray,
        statusObj => compareAsc(statusObj.date, date) === 0
      );

      if (status.length === 0) return <td />;
      if (status.length === 1) {
        return (
          <Status
            key={status[0].row}
            code={status[0].status}
            appTitle={appTitle}
            description={status[0].statusDescription}
            dateString={status[0].dateString}
            note={status[0].note}
            tableCell
            popOver
            id={`popover${status[0].row}`}
          />
        );
      }

      return (
        <Status
          statuses={status}
          appTitle={appTitle}
          tableCell
          popOver
          key={status[0].row}
          code='M'
          id={`popover${status[0].row}`}
        />
      );
    });
  }

  render() {
    const { appTitle } = this.props;
    return (
      <tr>
        <th style={{ whiteSpace: 'nowrap' }}>{appTitle}</th>
        {this.buildRow()}
      </tr>
    );
  }
}

TimelineRow.propTypes = {
  dateArray: PropTypes.string,
  appStatusArray: PropTypes.arrayOf,
  appTitle: PropTypes.string
};
TimelineRow.defaultProps = {
  dateArray: [],
  appStatusArray: [],
  appTitle: ''
};
