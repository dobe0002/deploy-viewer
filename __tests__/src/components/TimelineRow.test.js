/* eslint-disable react/jsx-filename-extension */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TimelineRow from '../../../src/components/TimelineRow';
import dateArray from '../../../__fixtures__/dateArray';

configure({ adapter: new Adapter() });

const { axe, toHaveNoViolations } = require('jest-axe');

expect.extend(toHaveNoViolations);

describe('Timeline component tests', () => {
  const sampleProps = {
    dateArray,
    appTitle: 'Training Hub',
    appStatusArray: [
      {
        row: 2,
        status: 'B',
        statusDescription: 'Bad Deployment',
        dateString: '11/01/2019',
        date: new Date(2019, 10, 1), // '2019-11-01T05:00:00.000Z',
        note: 'Ooopsie!'
      },
      {
        row: 4,
        status: 'G',
        statusDescription: 'Good deployment',
        dateString: '11/10/2019',
        date: new Date(2019, 10, 10), // '2019-11-10T06:00:00.000Z',
        note: 'Some stuff happened but it was ALL GOOD'
      }
    ]
  };
  test('Accessibility check', async () => {
    const wrapper = shallow(
      <TimelineRow
        dateArray={sampleProps.dateArray}
        appTitle={sampleProps.appTitle}
        appStatusArray={sampleProps.appStatusArray}
      />
    );
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });
  test('Snapshot test', () => {
    const wrapper = shallow(
      <TimelineRow
        dateArray={sampleProps.dateArray}
        appTitle={sampleProps.appTitle}
        appStatusArray={sampleProps.appStatusArray}
      />
    );
    expect(wrapper.find('tr')).toHaveLength(1);
    expect(wrapper.find('th')).toHaveLength(1);
    expect(wrapper.find('Status')).toHaveLength(2);
    expect(wrapper.find('td')).toHaveLength(30);
    expect(wrapper).toMatchSnapshot();
  });
});
