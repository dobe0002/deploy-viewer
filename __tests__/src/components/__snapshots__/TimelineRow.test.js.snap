// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline component tests Snapshot test 1`] = `
<tr>
  <th
    style={
      Object {
        "whiteSpace": "nowrap",
      }
    }
  >
    Training Hub
  </th>
  <Status
    appTitle="Training Hub"
    code="B"
    dateString="11/01/2019"
    description="Bad Deployment"
    icon={false}
    id="popover2"
    key="2"
    note="Ooopsie!"
    popOver={true}
    statuses={Array []}
    tableCell={true}
  />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <Status
    appTitle="Training Hub"
    code="G"
    dateString="11/10/2019"
    description="Good deployment"
    icon={false}
    id="popover4"
    key="4"
    note="Some stuff happened but it was ALL GOOD"
    popOver={true}
    statuses={Array []}
    tableCell={true}
  />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
</tr>
`;
